chronosphere/set_unknown.go

```diff
@@ -26,43 +26,43 @@ import (
 	"github.com/chronosphereio/terraform-provider-chronosphere/chronosphere/tfid"
 )
 
-// indexPath is used in path when indexing into a slice / map.
-const indexPath = "[]"
+// Paths used for indexing into lists/maps in the raw config.
+const (
+	indexPath       = "[]"
+	listEncodedPath = "[0]"
+)
 
 // dummyRef is used in dry run validation to populate fields that reference other entities
 // that don't exist yet.
 var dummyRef = tfid.Slug("dummy_value")
 
-// setUnknownReferences sets entity reference fields (those having the type tfid.ID) that
+type setUnknownParams struct {
+	rawConfig      cty.Value
```

Comment:
we don't use the `// required` linter in the terraform provider so we don't have any build time guarantees that all callers are sending in all values. Is it safe for a caller to only send a subset in? I looked at the code and i see that the only place we use this has everything covered, but if all the values are required then we should add validation in `setUnknown` and add some tests to validate.